tests/unit/test_utils.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2020 Ubuntu
+# See LICENSE file for licensing details.
+
+import os
+import pathlib
+import tempfile
+import unittest
+from uuid import uuid4
+
+import utils
+
+
+class TestUtils(unittest.TestCase):
+    def test_find_packages_by_indices(self):
+        nums = 3
+        expected_outputs = set()
+        returned_output = set()
+        # Generate a test file structure for testing
+        with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
We can put the golden testing folder with the unit test together. 
Then we can
1. create many differet test environment easily
2. do not create the test environment by code which is complicated and may be buggy.

Reply:
I was also thinking about that, but since you requested it, let me just do it.